### ISSUE TITLE
Assertion `Ptr != End && "dereferencing end() iterator"' failed.

### ISSUE THREAD

**Issue by sscalpone** [flang-compiler/f18-llvm-project#503]
```
% cat rl.f
C     ranlib/ranlib.f
      SUBROUTINE F
      GO TO 120
  120 END
```
```
% bbc rl.f |& head
bbc: /local/home/sjs/mirror/fir-dev-mirror/llvm/include/llvm/ADT/DenseMap.h:1241: llvm::DenseMapIterator<KeyT, ValueT, KeyInfoT, Bucket, IsConst>::value_type* llvm::DenseMapIterator<KeyT, ValueT, KeyInfoT, Bucket, IsConst>::operator->() const [with KeyT = long unsigned int; ValueT = Fortran::lower::pft::Evaluation*; KeyInfoT = llvm::DenseMapInfo<long unsigned int>; Bucket = llvm::detail::DenseMapPair<long unsigned int, Fortran::lower::pft::Evaluation*>; bool IsConst = false; llvm::DenseMapIterator<KeyT, ValueT, KeyInfoT, Bucket, IsConst>::pointer = llvm::detail::DenseMapPair<long unsigned int, Fortran::lower::pft::Evaluation*>*; llvm::DenseMapIterator<KeyT, ValueT, KeyInfoT, Bucket, IsConst>::value_type = llvm::detail::DenseMapPair<long unsigned int, Fortran::lower::pft::Evaluation*>]: Assertion `Ptr != End && "dereferencing end() iterator"' failed.
PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace.
Stack dump:
0.      Program arguments: bbc rl.f
 #0 0x0000561a4ad5aedd llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/local/home/sjs/mirror/install/bin/bbc+0x345dedd)
 #1 0x0000561a4ad58c14 llvm::sys::RunSignalHandlers() (/local/home/sjs/mirror/install/bin/bbc+0x345bc14)
 #2 0x0000561a4ad58d80 SignalHandler(int) (/local/home/sjs/mirror/install/bin/bbc+0x345bd80)
 #3 0x00007fd159d098a0 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x128a0)
 #4 0x00007fd158975f47 raise /build/glibc-2ORdQG/glibc-2.27/signal/../sysdeps/unix/sysv/linux/raise.c:51:0
 #5 0x00007fd1589778b1 abort /build/glibc-2ORdQG/glibc-2.27/stdlib/abort.c:81:0
```

**Comment by schweitzpgi** [comment]
Val, looks like it might be PFT/CFG related.

**Comment by vdonaldson** [comment]
f18 clause 5.3.3 explicitly allows branching to an END statement.

**Comment by schweitzpgi** [comment]
This part of the stack trace is helpful.
```
8  bbc                      0x000000010d315c13 llvm::DenseMapIterator<unsigned long long, Fortran::lower::pft::Evaluation*, llvm::DenseMapInfo<unsigned long long>, llvm::detail::DenseMapPair<unsigned long long, Fortran::lower::pft::Evaluation*>, false>::operator->() const + 195
9  bbc                      0x000000010ddf638d (anonymous namespace)::PFTBuilder::markBranchTarget(Fortran::lower::pft::Evaluation&, unsigned long long) + 141
10 bbc                      0x000000010ddfaa67 (anonymous namespace)::PFTBuilder::analyzeBranches(Fortran::lower::pft::Evaluation*, std::__1::list<Fortran::lower::pft::Evaluation, std::__1::allocator<Fortran::lower::pft::Evaluation> >&)::'lambda'(Fortran::parser::GotoStmt const&)::operator()(Fortran::parser::GotoStmt const&) const + 39
```

**Comment by vdonaldson** [comment]
In addition to this bridge fix, front end gitlab MR #161 has an additional fix that is needed for some cases.
